src/commands/templates.rs

```diff
@@ -441,7 +448,7 @@ impl List {
             table.set_header(header);
             table.load_preset(comfy_table::presets::ASCII_BORDERS_ONLY_CONDENSED);
 
-            for template in templates {
+            for template in self.filter_templates_with_tags(templates) {
```

Comment:
@itowlson 

w.r.t your first comment: Great catch. I think i can maneuver some of this code to avoid the empty table headers being printed as well to include a more informative "no templates matched with tags" message as you suggest. Thanks!

w.r.t your second comment: These are all great thoughts and absolutely appreciated. I had originally started to implement `TemplatesManager::list_with_tags` that does exactly as you suggest (with updated list results to reflect the behavior you mention). Happy to go that approach as its not too much more involved than what is already here.

Reply:
Updated to address some of the UI concerns.

# With no templates installed 
```
❯ spin templates list --tag http
You don't have any templates yet. Would you like to install the default set? [Y/n]
```

# With default set installed but no matching tag
```
❯ spin templates list --tag foo
No templates matched tags (12 templates skipped)
```

# The Spin new experience with tags
```
❯ spin new --tag http
Pick a template to start your application with:
> http-c (HTTP request handler using C and the Zig toolchain)
  http-empty (HTTP application with no components)
  http-go (HTTP request handler using (Tiny)Go)
  http-grain (HTTP request handler using Grain)
  http-php (HTTP request handler using PHP)
  http-rust (HTTP request handler using Rust)
  http-swift (HTTP request handler using SwiftWasm)
  http-zig (HTTP request handler using Zig)
  static-fileserver (Serves static files from an asset directory)
```

Reply:
BRIAN YOU DID IT

I was not expecting that as part of this PR - very cool!